### PR TITLE
Userlevel multithreading support for hotswap.

### PR DESCRIPTION
**Pull request by bcronje**
Close issue #173

### CODE REVIEW

File: userlevel/click.cc [hunk 1]
```diff
@@ -461,6 +426,58 @@ void do_set_affinity(pthread_t p, int cpu) {
...
+
+static bool
+hotswap_hook(Task *, void *)
+{
+    hotswap_thunk_router->set_foreground(false);
+    hotswap_router->activate(ErrorHandler::default_handler());
+    router->unuse();
+#if HAVE_MULTITHREAD
+    for (int i = 0; i < other_threads.size(); ++i) {
+        pthread_cancel(other_threads[i]);
```
Comment (bcronje) [inline userlevel/click.cc]: I dont like the pthread_cancel here much, but I found no way to signal the RouterThread drivers to stop before joining the threads. @kohler do you have a better suggestion to stop the drivers before joining?

**Comment by bcronje** [comment]
Eddie and Click team, this is a first attempt to enable support for userlevel multithreading support for hotswap. This seems to work fine in my tests, but feel free to suggest a better way to accomplish this.

**Comment by kohler** [comment]
Beyers!!

Thanks for pointing out this problem! I made an attempt to fix this in a different way. Rather than canceling the outstanding router threads, we “block” them, as in linuxmodule, and swap routers only when the block completes. a26ccfa1413a0371e7ad88ac302a070ba344899e

Does this work for you?

**Comment by kohler** [comment]
Beyers—any comments on this?

**Comment by bcronje** [comment]
@kohler Eddie, apologies for the slow response, got sidetracked with another project.

Awesome, I can confirm your implementation works perfectly and is indeed a cleaner way than my original PR. I'm closing this PR and issue #173 now.

Thanks again for looking into this, much appreciated!

**Comment by kohler** [comment]
Thank *you*!

**Pull request closed without merging**